### ISSUE TITLE
RFE: Thanos operator

### ISSUE THREAD

**Issue by aakarshg** [observatorium/thanos-receive-controller#36]
I was wondering if it makes sense to build a kubernetes operator that'll help with maintenance and configuration of the controller, and other components such as receive, querier and compactor. This will require a lot of resources, and I'm willing to help as much as I can. 

I'm just throwing it out here to stir up a conversation to see if what I'm proposing/requesting actually makes sense.

**Comment by brancz** [comment]
I think we’ll one day get there, but we’re still gathering operational experience for the various components and are trying to trade off what makes sense implementing in Thanos itself and what’s left to config management. The Thanos receive proposal explicitly offloaded the points currently handled by this controller to config management.

tl;dr one day but I personally think what we would currently come up with as APIs would likely be wrong and premature.
